backend/inspirehep/search/api.py

```diff
@@ -118,18 +124,58 @@ class Meta:
         doc_types = "_doc"
         default_filter = DefaultFilter(Q())
 
+    def query_for_superuser_or_users(self, query_string):
+        if not is_superuser_or_cataloger_logged_in():
+            user_query = Q(
+                IQ(query_string, self) & Q("term", _collections="Literature")
+            )
+            return self.query(user_query)
+        return self.query(IQ(query_string, self))
+
     def query_from_iq(self, query_string):
         """Initialize ES DSL object using INSPIRE query parser.
+
         :param query_string: Query string as a user would input in INSPIRE's search box.
         :type query_string: string
         :returns: Elasticsearch DSL search class
         """
-        if not is_superuser_or_cataloger_logged_in():
-            user_query = Q(
-                IQ(query_string, self) & Q("term", _collections="Literature")
+        search_query = self.query_for_superuser_or_users(query_string)
+        search_results = search_query.execute()
```

Comment:
am I understanding correctly that we're going to dispatch every query twice? there's no better way to do this?

Reply:
Yes, otherwise I have to do some really dirty hack in `execute` and I wanted to avoid that.

Reply:
@michamos I'll add a feature flag at the moment, I'd like first to see if it actually works as we expecting and If it works I'll find a better solution.